dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/DefaultAnalyticsTableService.java

```diff
@@ -172,6 +172,19 @@ public void create(AnalyticsTableUpdateParams params, JobProgress progress) {
     clock.logTime("Table update done: '{}'", tableType.getTableName());
   }
 
+  private List<AnalyticsTable> getAnalyticsTables(AnalyticsTableUpdateParams params) {
+    return tableManager.getAnalyticsTables(params).stream()
+        .map(analyticsTable -> withDistributedFlag(analyticsTable, params))
+        .toList();
+  }
+
+  private AnalyticsTable withDistributedFlag(
+      AnalyticsTable analyticsTable, AnalyticsTableUpdateParams params) {
+    analyticsTable.setTableDistributed(
+        params.isCitusExtensionEnabled() && analyticsTable.isTableTypeDistributed());
```

Comment:
Instead of having the Citus extension config info passed through the job `AnalyticsTableUpdateParams`, could we instead detect whether Citus is installed on startup, and expose it in the `AnalyticsTableSettings` interface. This will not change during runtime, so I don't think there is a need to pass it to every job.

Reply:
This was in the original implementation.  I will check for the possibility 👍